staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -255,10 +277,14 @@ func (a *OIDCAuthenticator) AuthenticateToken(value string) (user.Info, bool, er
 		if !emailVerified {
 			return nil, false, errors.New("email not verified")
 		}
-		username = claim
 	default:
-		// For all other cases, use issuerURL + claim as the user name.
```

Comment:
I would actually hoist the specified/unspecified/`-`/email/not-email logic up to `newAuthenticatorFromOIDCIssuerURL` so from this authenticator's perspective, it just prepends the specified prefixes. There's no reason for this authenticator to be aware of special flag values like `-`